src/Layers/DynamicMapLayer.js

```diff
@@ -87,19 +87,26 @@ export var DynamicMapLayer = RasterLayer.extend({
       }, this), 300);
     }, this);
 
-    var identifyRequest = this.identify().on(this._map).at(e.latlng);
+    var identifyRequest;
+    if (this.options.popup) {
+      identifyRequest = this.options.popup.on(this._map).at(e.latlng);
```

Comment:
i pushed a small change here to call `IdentifyFeatures.on()` dynamically even when a custom popup constructor option has been bound to the layer. this is beneficial for two reasons.

1. it avoids an error when `on()` wasn't chained prior.
2. even if the method was chained, we need to avoid passing a stale extent through in the request if the map was panned or zoomed after the layer was instantiated.